api/models/Campaign.js

```diff
@@ -31,8 +35,51 @@ const schema = new mongoose.Schema({
   msg_menu_signedup_gambit: String,
   msg_no_photo_sent: String,
 
+  // Mobile Commons Specific Fields.
+  mobilecommons_groups: {
+    doing: Number,
+    completed: Number,
+  },
+
 });
 
+/**
+ * For a given campaign, update its mobile commons group id,
+ * based on the mobile commons response.
+ * @param {Campaign} campaign    Campaign model to update.
+ * @param {string} status        Either completed or doing.
+ * @param {string} groupResponse XML string from mobile commons API.
+ */
+function setMobileCommonsGroup(campaign, status, groupResponse) {
+  const scope = campaign;
+  const parsedGroup = JSON.parse(parser.toJson(groupResponse));
+  // If the group name is available...
+  if (parsedGroup.response.success === 'true') {
+    // Save newly created group id to this campaign.
+    const groupId = parsedGroup.response.group.id;
+    scope.mobilecommons_groups[status] = groupId;
+  }
+}
+
+/**
+ * Create Doing/Completed Mobile Commons Groups to support Mobile Commons broadcasting.
+ * @see https://github.com/DoSomething/gambit/issues/673
+ */
+schema.methods.createMobileCommonsGroups = function () {
+  const campaign = this;
+  const prefix = `env=${process.env.NODE_ENV}
+                  campaign_id=${campaign._id}
+                  run_id=${campaign.current_run}`;
+
+  // Create mobile commons group with custom name based on this campaign
+  mobilecommons.createGroup(`${prefix} status=doing`)
+  .then(doingGroup => setMobileCommonsGroup(campaign, 'doing', doingGroup))
+  .then(() => mobilecommons.createGroup(`${prefix} status=completed`))
+  .then(completedGroup => setMobileCommonsGroup(campaign, 'completed', completedGroup))
+  .then(() => campaign.save())
```

Comment:
Campaign is definitely updated, I think this is related to the problem we talked about below with you not being able to pull data.